ci-operator/templates/openshift/installer/cluster-launch-installer-nested-virt-tests.yaml

```diff
@@ -104,264 +104,114 @@ objects:
-        # to create the image-family see here: https://github.com/ironcladlou/openshift4-libvirt-gcp#images
-        gcloud compute instances create "${INSTANCE_PREFIX_TEST}" \
-          --image-family openshift4-libvirt \
-          --zone "${GOOGLE_COMPUTE_ZONE}" \
-          --machine-type n1-standard-8 \
-          --min-cpu-platform "Intel Haswell" \
-          --boot-disk-type pd-ssd \
-          --boot-disk-size 256GB
-        LD_PRELOAD=/usr/lib64/libnss_wrapper.so gcloud compute scp \
-          --quiet \
-          --project "${GOOGLE_PROJECT_ID}" \
-          --zone "${GOOGLE_COMPUTE_ZONE}" \
-          --recurse /home/packer/kubevirt packer@"${INSTANCE_PREFIX_TEST}":~/
-        LD_PRELOAD=/usr/lib64/libnss_wrapper.so gcloud compute scp \
-          --quiet \
-          --project "${GOOGLE_PROJECT_ID}" \
-          --zone "${GOOGLE_COMPUTE_ZONE}" \
-          --recurse "${HOME}"/kubevirt-tests.sh packer@"${INSTANCE_PREFIX_TEST}":~/kubevirt/kubevirt-tests.sh
-        LD_PRELOAD=/usr/lib64/libnss_wrapper.so gcloud compute scp \
-          --quiet \
-          --project "${GOOGLE_PROJECT_ID}" \
-          --zone "${GOOGLE_COMPUTE_ZONE}" \
-          --recurse "${HOME}"/run-tests.sh packer@"${INSTANCE_PREFIX_TEST}":~/run-tests.sh
-        # we don't run openshift-tests, but might in the future
-        LD_PRELOAD=/usr/lib64/libnss_wrapper.so gcloud compute scp \
-          --quiet \
-          --project "${GOOGLE_PROJECT_ID}" \
-          --zone "${GOOGLE_COMPUTE_ZONE}" \
-          --recurse "${HOME}"/openshift-tests packer@"${INSTANCE_PREFIX_TEST}":~/openshift-tests
-        # Now wait for setup container to finish cluster launch, then copy kubeconfig to nested-virt-enabled gcp
-        # where tests will run.
-        while true; do
-          if [[ ! -f "${HOME}"/setup-success ]]; then
-            sleep 15 & wait
-            continue
-          fi
-          break
-        done
-        LD_PRELOAD=/usr/lib64/libnss_wrapper.so gcloud compute scp \
-          --quiet \
-          --project "${GOOGLE_PROJECT_ID}" \
-          --zone "${GOOGLE_COMPUTE_ZONE}" \
-          --recurse "${HOME}"/kubeconfig packer@"${INSTANCE_PREFIX_TEST}":~/kubeconfig
+        # set up cloud-provider-specific env vars
+        export KUBE_SSH_BASTION="$( oc --insecure-skip-tls-verify get node -l node-role.kubernetes.io/master -o 'jsonpath={.items[0].status.addresses[?(@.type=="ExternalIP")].address}' ):22"
+        export KUBE_SSH_KEY_PATH=/tmp/cluster/ssh-privatekey
+        if [[ "${CLUSTER_TYPE}" == "gcp" ]]; then
+          export GOOGLE_APPLICATION_CREDENTIALS="${GCP_SHARED_CREDENTIALS_FILE}"
+          export KUBE_SSH_USER=core
+          mkdir -p ~/.ssh
+          cp /tmp/cluster/ssh-privatekey ~/.ssh/google_compute_engine || true
+          export TEST_PROVIDER='{"type":"gce","region":"us-east1","multizone": true,"multimaster":true,"projectid":"openshift-gce-devel-ci"}'
+        elif [[ "${CLUSTER_TYPE}" == "aws" ]]; then
```

Comment:
To my knowledge nested virtualization on AWS is supported only with emulation and that should be dead slow? Am I wrong here?

Reply:
right, hm.. but, the test container is running in gcp (I believe the api.ci cluster runs in gce?) and that is what has to have nested-virt enabled, right? or does the external cluster require nested-virt?  anyways, we can switch to only test in gcp, but I was thinking you'd want to eventually test kubevirt against clusters in all supported cloud providers?